plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java

```diff
@@ -311,15 +318,21 @@ private void addContainer(IDeploymentConfig dc, String workspaceID, String image
         dc.getContainer(containerName).setImagePullPolicy(IMAGE_PULL_POLICY_IFNOTPRESENT);
 
         ModelNode dcFirstContainer = ((DeploymentConfig)dc).getNode().get("spec").get("template").get("spec").get("containers").get(0);
+        
+        boolean isDevMachine = isDevMachine(exposedPorts);
+        // Run non-dev machine as root user
+        String UID = isDevMachine ? UID_USER : UID_ROOT;
```

Comment:
@l0rd thanks for review. For now the logic is the following - all dev mashines are run as user (UID 1000) and all non-dev machines as root (UID 0). Basically, I just checked what UID is used for MySQL-Java stack in by running Che without OpenShift.  I will try to update PR with the logic of getting user from image config